contracts/src/v0.8/keystone/KeystoneFeedsConsumer.sol

```diff
@@ -2,27 +2,96 @@
 pragma solidity ^0.8.19;
 
 import {IReceiver} from "./interfaces/IReceiver.sol";
+import {ConfirmedOwner} from "../shared/access/ConfirmedOwner.sol";
 
-contract KeystoneFeedsConsumer is IReceiver {
-  event MessageReceived(bytes32 indexed workflowId, address indexed workflowOwner, uint256 nReports);
+contract KeystoneFeedsConsumer is IReceiver, ConfirmedOwner {
+  event MessageReceived(bytes10 indexed workflowName, address indexed workflowOwner, uint256 nReports);
   event FeedReceived(bytes32 indexed feedId, uint256 price, uint64 timestamp);
 
-  constructor() {}
+  error UnauthorizedSender(address sender);
+  error UnauthorizedWorkflowOwner(address workflowOwner);
+  error UnauthorizedWorkflowName(bytes10 workflowName);
+
+  constructor() ConfirmedOwner(msg.sender) {}
 
   struct FeedReport {
-    bytes32 FeedID;
+    bytes32 FeedId;
     uint256 Price;
     uint64 Timestamp;
   }
 
-  function onReport(bytes32 workflowId, address workflowOwner, bytes calldata rawReport) external {
-    // TODO: validate sender and workflowOwner
+  mapping(bytes32 feedId => uint256 price) internal s_prices;
+  address[] internal allowedSenders;
+  address[] internal allowedWorkflowOwners;
+  bytes10[] internal allowedWorkflowNames;
+
+  function setConfig(address[] calldata _allowedSenders, address[] calldata _allowedWorkflowOwners, bytes10[] calldata _allowedWorkflowNames) external onlyOwner {
+    allowedSenders = _allowedSenders;
+    allowedWorkflowOwners = _allowedWorkflowOwners;
+    allowedWorkflowNames = _allowedWorkflowNames;
```

Comment:
You want to set this to a single hashed storage value to save storage read costs.

Reply:
This can be a mapping if you want to allow multiple permutations (you probably do).

Reply:
This could be a `mapping(bytes32 => bool)`. This should be sufficient; no need for a hash.

Reply:
If I use a mapping, how do I delete old values?

Reply:
I'm guessing that I need to store a map and a list as well. Let me know if you had something else in mind.